lib/shared/drive-constraints.js

```diff
@@ -208,3 +208,85 @@ exports.isDriveValid = (drive, image) => {
 exports.isDriveSizeRecommended = (drive, image) => {
   return _.get(drive, [ 'size' ], UNKNOWN_SIZE) >= _.get(image, [ 'recommendedDriveSize' ], UNKNOWN_SIZE);
 };
+
+const feedbackMessages = {
+  NOTREC: 'Not Recommended',
+  TOOSMALL: 'Too Small For Image',
+  LOCKED: 'Locked',
+  SYSTEM: 'System Drive',
+  SOURCE: 'Drive Contains Image'
+};
+const FEEDBACK_ERROR = 'error';
+const FEEDBACK_WARNING = 'warning';
+
+/**
+ * @summary Get drive feedback in an object, if any
+ * @function
+ * @public
+ *
+ * @description
+ * If the drive-image combination results in a warning or error, this function
+ * returns an object containing the feedback type ('error', 'warning') and the
+ * feedback message.
+ *
+ * @param {Object} drive - drive
+ * @param {Object} image - image
+ * @returns {Object} feedback object
+ *
+ * @example
+ * const drive = {
+ *   device: '/dev/disk2',
+ *   name: 'My Drive',
+ *   size: 4000000000
+ * };
+ *
+ * const image = {
+ *   path: 'rpi.img',
+ *   size: 2000000000
+ *   recommendedDriveSize: 4000000000
+ * });
+ *
+ * const { type, message } = DriveConstraints.getDriveFeedback(drive, image);
+ *
+ * if (type === 'warning') {
+ *   // still allow flashing
+ * } else if (type === 'error') {
+ *   // disallow flashing
+ * }
+ */
+exports.getDriveFeedback = (drive, image) => {
```

Comment:
Yeah, I'm guessing "feedback" was chosen because Benedict was coming at this from a "visual feedback" point of view? I don't think `.getDriveImageFlashingMessage` works either, because it sounds like it might be the message displayed _during_ the flashing process.
What about something like `.getDriveImageCompatibilityStatus` ? (since it's effectively telling you whether a given image is 'allowed' to be flashed to a given drive, and the reason why it's not otherwise)

Also, perhaps we should check that this function works with a 'null' image supplied, so that we don't have to revisit this function if we swap around the GUI order to select the drive before the image?